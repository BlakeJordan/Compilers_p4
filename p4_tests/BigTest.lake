int g(bool a) {
    return;
}

int f() {
    bool a;
    a = g;
}

void foo() {
    return;
}

int main() {
    int c;
    int a;
    int b;
    bool d;
    int @ z;
    z = true;
    z = 1;
    b = 1;
    a = 2;
    c = foo();
    c = d * b;
    c = a/b;
    d = a + b;
    d = true;
    d = false;
    d = a < b;
    d = a > b;
    d = a <= b;
    d = a >= b;
    f();
    f(45, true);
    g(d);
    g();
    g(45, true);
    write z;
    read z;
    d = b == c;
    d = a != c;
    return true;
}