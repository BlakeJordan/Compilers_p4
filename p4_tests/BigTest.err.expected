2,5: Missing return value
7,9: Invalid assignment operand
20,7: Invalid assignment operation
21,7: Invalid assignment operation
24,7: Invalid assignment operation
25,9: Arithmetic operator applied to invalid operand
27,7: Invalid assignment operation
35,5: Function call with wrong number of args
37,5: Function call with wrong number of args
38,5: Function call with wrong number of args
39,11: Attempt to write a raw pointer
40,10: Attempt to read a raw pointer
43,12: Bad return value
Type checking failed